src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/AddedValidationPreparer.java

```diff
@@ -1,59 +1,62 @@
-import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.PaymentStatus;
-import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
-
-@Component
-public class AddedValidationPreparer implements PreSubmitCallbackHandler<AsylumCase> {
-
-    private final boolean isfeePaymentEnabled;
-
-    public AddedValidationPreparer(
-        @Value("${featureFlag.isfeePaymentEnabled}") boolean isfeePaymentEnabled
-    ) {
-        this.isfeePaymentEnabled = isfeePaymentEnabled;
-    }
-
-    public boolean canHandle(
-        PreSubmitCallbackStage callbackStage,
-        Callback<AsylumCase> callback
-    ) {
-
-        requireNonNull(callbackStage, "callbackStage must not be null");
-        requireNonNull(callback, "callback must not be null");
-
-        return (callbackStage == PreSubmitCallbackStage.ABOUT_TO_START)
-               && Event.PAYMENT_APPEAL == callback.getEvent()
-               && isfeePaymentEnabled;
-    }
-
-    public PreSubmitCallbackResponse<AsylumCase> handle(
-        PreSubmitCallbackStage callbackStage,
-        Callback<AsylumCase> callback
-    ) {
-        if (!canHandle(callbackStage, callback)) {
-            throw new IllegalStateException("Cannot handle callback");
-        }
-
-        AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
-
-        final PreSubmitCallbackResponse<AsylumCase> asylumCasePreSubmitCallbackResponse = new PreSubmitCallbackResponse<>(asylumCase);
-
-        if (asylumCase.read(PAYMENT_STATUS, PaymentStatus.class).equals(Optional.of(PaymentStatus.PAID))) {
-            asylumCasePreSubmitCallbackResponse.addError("You have already paid for this appeal.");
-        }
-
-        return asylumCasePreSubmitCallbackResponse;
-    }
-}
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.presubmit.payment;
+
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.PAYMENT_STATUS;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.REFUND_CONFIRMATION_APPLIED;
+
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.PaymentStatus;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.YesOrNo;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
+
+@Component
+public class AddedValidationPreparer implements PreSubmitCallbackHandler<AsylumCase> {

```

Comment:
@ikirsanov, are you using a different formatting? Why is the whole file has diff?

Reply:
Done